src/Symfony/Component/HttpFoundation/Cookie.php

```diff
@@ -22,13 +22,14 @@ class Cookie
     public const SAMESITE_LAX = 'lax';
     public const SAMESITE_STRICT = 'strict';
 
-    protected $name;
-    protected $value;
-    protected $domain;
-    protected $expire;
-    protected $path;
-    protected $secure;
-    protected $httpOnly;
+    protected string $name;
+    protected ?string $value;
+    protected ?string $domain;
+    protected int $expire;
+    protected string $path;
+    protected ?bool $secure;
+    protected bool $httpOnly;
```

Comment:
this must be reverted IMHO, as it's a BC break, isn't it? 🤔

Reply:
The constructor's signature and setter methods are type-hinted, so there wouldn't be any BC. But an extended class using these parameters with different types would lead to a BC. I considered this as an "unlikely BC" and a "misuse" of class attributes. But anyway, it could be a BC.

Let me know what's your guideline here and I will fix it.

Reply:
ping @nicolas-grekas